mindsay_sdk/client.py

```diff
@@ -131,34 +154,44 @@ def get_user_nodes(self) -> List[Dict[str, Any]]:
         response.raise_for_status()
         return response.json()
 
-    def get_user_node(self, user_node_record_id: int) -> dict:
+    def get_user_node(self, user_node_record_id: int) -> Dict[str, Any]:
         """Returns the required user node"""
         response = self.get(f'/user_nodes/{user_node_record_id}')
         response.raise_for_status()
         return response.json()
 
-    def get_intent(self, intent_record_id: int) -> dict:
+    def get_intent(self, intent_record_id: int) -> Dict[str, Any]:
         """Returns the intent matching the given id"""
         response = self.get(f'intents/{intent_record_id}')
         response.raise_for_status()
         return response.json()
 
-    def deploy_intent(self, intent_record_id: int):
+    def deploy_intent(self, intent_record_id: int) -> Dict[str, Any]:
         """Deploy the intent matching the given id"""
         response = self.put(f'intents/{intent_record_id}/deploy')
         response.raise_for_status()
         return response.json()
 
-    def update_entity(self, entity_record_id: int, entity: dict):
+    def update_entity(self, entity_record_id: int, entity: dict) -> Dict[str, Any]:
         """Update the entity matching the given id with the given entity object"""
         response = self.put(f'entities/{entity_record_id}', json=entity)
         response.raise_for_status()
         return response.json()
 
-    def get(self, url, **kwargs):
+    def get(self, url, **kwargs) -> requests.Response:
         """Perform a GET request using the Mindsay base URL"""
         return super().get(self.base_url + url, **kwargs)
 
+    def get_all(self, object_type: str) -> List[dict]:
+        """Gather detailed information for all the objects of a given type (e.g. 'answers')"""
```

Comment:
Where can we get documentation on the different object types available? Is there an endpoint for this?

Reply:
Nope and that's why I'm not a big fan of asking the users to provide the resource name in parameter.
Maybe we can keep this function private and expose more explicit helpers for each resource like `get_all_user_nodes` or `get_entire_user_nodes` ?

Reply:
I agree with this, a list comprehension to get all user nodes with full info seems like a common enough idiom to warrant a helper method dedicated to this.

@Anogio do you intend to work on this or can I continue working on this PR?

Reply:
You can continue working on it if you want :) However I've been using this method a lot in my clone script, that's why I wrote it instead of using the comprehension every time (though it's also because my version had a progress bar)

Reply:
This is why this process is problematic... If I change this I will break your script, but if you hadn't told me I would have had no way to know this beforehand! We need a system to ensure we don't break scripts in another repository, otherwise we will realise in 3 months that scripts are broken without knowing what happened

Reply:
The process seems quite simple : scripts should never use SDK functions that have not been merged on master.
If you need this kind of helper, it's easy to write them in your script and open a PR to include them in the SDK if needed :)

Reply:
No it's alright I'm not using it from the SDK !
It's defined directly in the script, I just wanted to say I find it useful but if you don't think we need it it won't break my script